doc/requirements.txt

```diff
@@ -0,0 +1,7 @@
+numpy
+scipy
+matplotlib
+sphinx
+sphinxcontrib-apidoc
```

Comment:
Is this the preferred way of doing this? Or is it better to run apidoc manually? Does this work with RTD?

Reply:
The `.readthedocs.yml` file in the root points to this file, and RTD will install these dependencies before trying to build the documentation. I usually use `tox` to create and manage environments for running tests and building the documentation, then you set the requirements in `pyproject.toml` with everything else. But RTD can't work with that, so then you get either duplication or a hack.

Reply:
As for apidoc, RTD used to not run it, and you needed an ugly hack with a function in `conf.py` that hooked a Sphinx event and ran it from Python by calling an internal interface that differed depending on the version of apidoc you were using. It could be that RTD does run apidoc now, as it was an oft-requested featured, but you'd need it locally anyway, so when I found this plugin I thought it was a good idea.

I considered trying to set up RTD for my fork to see if it worked, but RTD doesn't take the organisation name into account and I'm worried about blocking transiflow.readthedocs.io. So it's probably better if you set it up, and I'm happy to help debug any issues then.